doc/ipv6/ipv6_link_local.md

```diff
@@ -0,0 +1,378 @@
+
+These addresses are restricted to a particular interface and are only used for addressing on that interface. They are typically used for automatic address configuration and neighbor discovery protocol. Link-local addresses can be used to reach the neighboring nodes attached to the same interface. These interfaces do not need a globally unique address to communicate. L3 routers do not forward IPv6 packets with link-local source or destination addresses. Every IPv6 enabled interface has at least one link-local unicast address.
+
+### 2.2.1 IPv6 Global Command
+Global IPv6 enable/disable command loops through all eligible interfaces and enables/disables the IPv6 'use-link-local-only' configuration. It is not a configuration knob but an action command. Eligible interface criteria is mentioned in section below.
+
+By default, the IPv6 mode is disabled on an interface excepting the out-of-band interface (eth0) and the linux kernel loopback interface (lo).
+
+### 2.2.1 IPv6 use-link-local-only
+The 'use-link-local-only' configuration is enabled on an interface to enable IPv6 mode on the interface as well as to enable IPv6 auto link-local generation, when there are not any explicitly configured addresses on the interface.
+
+In other words, the IPv6 mode on an interface is enabled either when addresses (global or link-local) are manually configured (or) when IPv6 'use-link-local-only' mode is enabled.
+
+Similarly, the IPv6 mode on an interface is disabled when IPv6 'use-link-local-only' mode is disabled with no manually configured addresses on the interface (or) when all addresses are unconfigured on the interface with IPv6 'use-link-local-only' disabled.
+
+User can configure IPv6 'use-link-local-only' on an interface that satisfies the following criteria.
+
+```
+- L2 interface that is not part of a Port-Channel or a VLAN.
+- L3 interface (Ethernet, VLAN, Port-Channel, Loopback)
+```
+
+An Ethernet interface is not allowed to be a member of a Port-Channel or a VLAN if 'use-link-local-only' configuration is enabled on the interface.
+A Port-Channel is not allowed to be a member of a VLAN if 'use-link-local-only' configuration is enabled on the interface.
+
+# 3 Design
+
+## 3.1 Overview
+The following components are modified to support IPv6 link local addresses.
+- Route ORCH agent
+- Neighbor ORCH agent
+- CONFIG_DB, APP_DB, ASIC_DB and STATE_DB updates
+- Interface manager
+
+## 3.2 DB Changes
+### 3.2.1 CONFIG DB
+To support enabling/disable of IPv6 auto link-local address generation on an interface, a new key-value pair is defined as part of CONFIG_DB under INTERFACE table. The default value is "disable".
+
+    "INTERFACE": {
+        "Ethernet24": {
+            "ipv6_use_link_local_only": "disable"

```

Comment:
Do we throw an error if user tries to configure global Ipv6 address when ipv6_use_link_local_only = true for L3 interface?

Reply:
ipv6_use_link_local_only config mode is in effect only in the absence of configured IPv6 global addresses on the interface. In the presence of global addresses, the link-local is enabled implicitly anyways.

Reply:
Even if we introduce 'ipv6_enable' field, it is implicitly enabled upon first global IPv6 configuration. it is aligned with what we exactly do to kernel in the cfgmgr.

By default Ipv6 is disabled on the interface, after few Ipv6 address configuration, if we need to go to the default state i.e ipv6 disable on interface, do we have any field? ipv6_use_link_local_only - disable should not disable the Ipv6 functionality on the interface or do we need to assume in the cfgmgr that when no IPv6 present on the interface, disable the IPv6 functionality? IMO, we can have the field name based on exactly what we do in the backend to avoid any confusion.